semantic/graph.go

```diff
@@ -499,7 +499,7 @@ type ArrayExpression struct {
 
 	Elements []Expression
 
-	typ MonoType
+	Type MonoType
```

Comment:
Why the change to make it public? The expression interface has the `TypeOf` method?

Reply:
I think `resolveValue` needs to be refactored, but it's in a different package and so I don't have access to modify it. I am not sure `resolveValue` is needed anymore and, in my opinion, it is a bit of a code smell that it converts a value into a semantic node.